### ISSUE TITLE
Incompatible With Anki 2.1.24

### ISSUE THREAD

**Issue by silentcicada1** [glutanimate/review-heatmap#67]
#### Problem description

*Please describe the issue concisely in here. In case of an error: Walk us through the steps you took to get there. What happened? What did you expect to happen?*
Not compatible With Anki 2.1.24

Error
An error occurred. Please start Anki while holding down the shift key, which will temporarily disable the add-ons you have installed.
If the issue only occurs when add-ons are enabled, please use the Tools > Add-ons menu item to disable some add-ons and restart Anki, repeating until you discover the add-on that is causing the problem.
When you've discovered the add-on that is causing the problem, please report the issue on the add-on support site.
Debug info:
Anki 2.1.24 (d3964603) Python 3.8.0 Qt 5.13.1 PyQt 5.14.1
Platform: Mac 10.15.1
Flags: frz=True ao=True sv=2
Add-ons, last update check: 2020-04-03 22:16:45
Add-ons possibly involved: ⁨Review Heatmap, Deck name in title 21⁩

Traceback (most recent call last):
  File "aqt/main.py", line 468, in loadCollection
  File "aqt/main.py", line 604, in moveToState
  File "aqt/main.py", line 611, in _deckBrowserState
  File "/Users/rachelstandifird/Library/Application Support/Anki2/addons21/699175524/__init__.py", line 44, in wrapped
    orig(*args, **kwargs)
  File "</Applications/Anki 2.1.24beta3.app/Contents/MacOS/decorator.pyc:decorator-gen-146>", line 2, in show
  File "anki/hooks.py", line 640, in decorator_wrapper
  File "anki/hooks.py", line 631, in repl
  File "aqt/deckbrowser.py", line 54, in show
  File "aqt/deckbrowser.py", line 110, in _renderPage
  File "aqt/deckbrowser.py", line 117, in __renderPage
  File "</Applications/Anki 2.1.24beta3.app/Contents/MacOS/decorator.pyc:decorator-gen-210>", line 2, in _renderStats
  File "anki/hooks.py", line 640, in decorator_wrapper
  File "anki/hooks.py", line 637, in repl
  File "/Users/rachelstandifird/Library/Application Support/Anki2/addons21/review_heatmap/views.py", line 92, in deckbrowserRenderStats
    html = ret + hmap.generate(view="deckbrowser")
  File "/Users/rachelstandifird/Library/Application Support/Anki2/addons21/review_heatmap/heatmap.py", line 88, in generate
    data = self.activity.getData(limhist=limhist, limfcst=limfcst)
  File "/Users/rachelstandifird/Library/Application Support/Anki2/addons21/review_heatmap/activity.py", line 69, in getData
    return self._getActivity(**self._reviewsData(time_limits))
  File "/Users/rachelstandifird/Library/Application Support/Anki2/addons21/review_heatmap/activity.py", line 168, in _reviewsData
    "forecast": self._cardsDue(start=self.today,
  File "/Users/rachelstandifird/Library/Application Support/Anki2/addons21/review_heatmap/activity.py", line 325, in _cardsDue
    return self.col.db.all(cmd, today=self.col.sched.today)
TypeError: all() got an unexpected keyword argument 'today'

#### Checklist

*Please replace the space inside the brackets with an **x** if the following items apply:*

 - [x] I've restarted Anki to see if it helps
 - [x] I've verified that I use the latest version of the add-on by redownloading it from AnkiWeb
 - [x] I've verified that I use the latest version of Anki by checking at https://apps.ankiweb.net#download
 - [ ] I've tried to disable other add-ons to see if there are any interactions present
 - [ ] My issue disappears when I hold shift while starting Anki.
 - [x] I've checked if anyone else reported this problem before by looking through the issue reports. I also checked to see if there is a section about known issues in the add-on description, documentation, or README.
 

#### Information about your Anki set-up

*Please fill out the section corresponding with your Anki version:*

##### If you are using Anki 2.1

*Please open Anki, go to Help → About, click on "Copy Debug Info", and paste the result between the backticks below (if the button does not appear you are using an older version of Anki 2.1 and will need to update first):*

```

```

#####  If you are using Anki 2.0

*__ONLY IF YOU ARE USING ANKI 2.0__: Please open Anki, go to Help → About, find the two lines starting with `Version:` and `Qt`, and copy-paste them between the backticks below:*

```

```

*Please fill in details about your operating system (Windows/macOS/Linux, which version):*

- OS:
- version:

*Please open Anki, go to Tools → Add-ons, take a screenshot of your installed add-ons, and paste it below:*

#### Error message (if any)

*If you've received an error message, please copy and paste it between the backticks below:*

```python

```

**Comment by chris360KS** [comment]
Can confirm this is happening in my environment as well (AnkiVersion 2.1.24 (d3964603)
Python 3.8.0 Qt 5.13.1 PyQt 5.14.1 on Mac OS X). 

Removing the keyword argument of `today=` on line 325 of `activity.py` with the following stopped the crash:
`return self.col.db.all(cmd, self.col.sched.today)`

**Comment by glutanimate** [comment]
Thanks again for the reports, guys. Due to changes on Anki's side, this should no longer be an issue in 2.1.24 stable. But I've also pushed a change to fortify the add-on against future deprecations: https://github.com/glutanimate/review-heatmap/commit/4e981a7452f89c244d5b08b0239eac0711eac618

Concerning 2.1.24 compatibility in general: There are a couple more minor compatibility issues which I've also addressed since (05e7ad6cbb19939d38168d966f85ebc3a6306683 & e09c6075f48e55491f4dec93a97b14947482d6cc). They will find their way into the next RH beta update which should becoming out somewhere over the next couple of days.

Closing this report for now as the original issue is fixed.
